src/App/integrationTests/app.lastRevisionNumbersForSyncing.test.js

```diff
@@ -0,0 +1,130 @@
+import '@testing-library/jest-dom/extend-expect'
+import { rest } from 'msw'
+import React from 'react'
+import userEvent from '@testing-library/user-event'
+
+import {
+  screen,
+  renderAuthenticatedOnline,
+  mockMermaidApiAllSuccessful,
+  within,
+  waitForElementToBeRemoved,
+} from '../../testUtilities/testingLibraryWithHelpers'
+import { getMockDexieInstancesAllSuccess } from '../../testUtilities/mockDexie'
+import { initiallyHydrateOfflineStorageWithMockData } from '../../testUtilities/initiallyHydrateOfflineStorageWithMockData'
+import App from '../App'
+
+const apiBaseUrl = process.env.REACT_APP_MERMAID_API
+
+test('When a sync pull responds with sync errors for 403 (user denied pulling data), the app will reset the last revision numbers for the data type and project to ensure the user can pull fresh data if they are given permission again', async () => {
```

Comment:
it is unintuitive! We need some api docs. Last revision numbers tell the back end how much new data we need. So the way it is set up now, when the app gets a for for a type of data for a project, it deletes the associated browser records. If we dont also delete the last revision number, our next pull will be empty because we are sending a last revision number that indicates that we are up to date

Reply:
deleting the last revision number associated with the 403 means we are telling the api to send us everything, not jsut new-to-us stuff

Reply:
hmm ok thx for explaining. This would be good to have in the future error section of the api docs @saanobhaai 

I can help get this started, would be good to have dedicated time to this next week!

Reply:
I just posted this in our team lead channel, but it is relavent here: https://jamison.dance/03-16-2019/starting-a-new-job-and-noticing-things

Reply:
thx haha will continue to point things out as I see them!